content/en/platform/corda/5.2/deploying-operating/keys/_index.md

```diff
@@ -0,0 +1,198 @@
+
+In both cases, Corda stores the master wrapping key settings in the `corda.crypto` configuration section.
+For information about the configuration fields in this section, see [corda.crypto]({{< relref "../config/fields/crypto.md" >}}).
+For information about how to retrieve the current values of a configuration section, see [Retrieving Current Configuration Values]({{< relref "../config/dynamic.md#retrieving-current-configuration-values" >}}).
+
+## Rotating Wrapping Keys
+
+R3 recommend changing the wrapping keys frequently to limit the probability of a compromised key being in use.
+Corda supports the following key rotation:
+
+* [Master Wrapping Key](#rotating-master-wrapping-key)
+* [Managed Wrapping Keys](#rotating-managed-wrapping-keys)
+
+### Rotating Master Wrapping Key
+
+Because the master key is not managed by Corda, you must first change the default key in the source system.
+Corda must then re-encrypt all of the keys that were wrapped with the old master key.
+To rotate the master wrapping key, do the following:
+
+1. Add the new master key to the source environment. For more information, see [Configuring the Master Wrapping Key](#configuring-the-master-wrapping-key).
+2. Add the new key to the `corda.crypto` configuration section, setting it as the default key. The [Setting Configuration Fields Dynamically]({{< relref "../config/dynamic.md#setting-configuration-fields-dynamically" >}}) section describes how to update configuration values. For example, to add a new master key, with the alias `master2`, [generated by the Default Secrets Service]({{< relref "../config/secrets.md#default-secrets-service" >}}):
+
+   {{< tabs >}}
+   {{% tab name="Bash"%}}
+   ```shell
+   curl -k -u $REST_API_USER:$REST_API_PASSWORD -X PUT -d '{"section":"corda.crypto", "version":"1", "config": \
+    "{\"caching\ :\"expireAfterAccessMins\":{\"default\":60},\"maximumSize\" :{\"default\":10000}},\"hsm\": \
+    {\"defaultWrappingKey\":\"master2\",\"retrying\": {\"attemptTimeoutMills\":20000, \"maxAttempts\":3}, \"wrappingKeys\": \
+    [{\"alias\":\"master1\", \"passphrase\": {\"configSecret\": {\"encryptedSecret\" :\"ZAmGcppwqUNsViSE06b801DVARbD9wT7SZnYP03WKZFOnC9KTsII04pWWq5Je2CUZdP+xIsHFUO3RkZVwhwnyQhWoxcW6JEo\"}},\"salt\": \
+    {\"configSecret\":{\"encryptedSecret\":\"cvNVJ4D7fNGm59Uh3Tlhi0W3pToDxl7vykUdZaxEuEPGUMdmu7G+xTEfl0CcC2R9aWl6FgrzmX5x+qCGNElUi/PBA40EGjEU\"}}}, \
+    {\"alias\":\"master2\",\"passphrase\":{\"configSecret\": {\"encryptedSecret\":\"gzXRjE4AoZSW8GQhcqbw0shQfwWGy5O8alNDqXaqTUzyUnLBVUXLwhv+uAEZzecQixiaEaNtfOvuftYW9NjwwRyf6m4KZIag\"}},\"salt\": \
+    {\"configSecret\":{\"encryptedSecret\":\"flvAnf5G/aRmDIaahXHjftEpc0odufhzZsuiipxtQTFV+pccB+RHnIqle26jjDlDUBfn3HPbVezKhpRHL/NP5M+cX/rU15DE\"}}}]}, \
+    \"retrying\":{\"maxAttempts\": {\"default\": 3},\"waitBetweenMills\":{\"default\":[200]}}}}","schemaVersion": {"major":1, "minor":0}}' \
+    "$REST_API_URL/config"
```

Comment:
I can't get this command working.
But this one works fine (with different secrets):
```kotlin
curl -k -u admin:admin  -X 'PUT' \
  'https://localhost:8888/api/v5_2/config' \
  -d '{
  "section": "corda.crypto",
  "config": "{\"caching\":{\"expireAfterAccessMins\":{\"default\":60},\"maximumSize\":{\"default\":10000}},\"hsm\":{\"defaultWrappingKey\":\"master2\",\"retrying\":{\"attemptTimeoutMills\":20000,\"maxAttempts\":3},\"wrappingKeys\":[{\"alias\":\"master1\",\"passphrase\":{\"configSecret\":{\"encryptedSecret\":\"ZAmGcppwqUNsViSE06b801DVARbD9wT7SZnYP03WKZFOnC9KTsII04pWWq5Je2CUZdP+xIsHFUO3RkZVwhwnyQhWoxcW6JEo\"}},\"salt\":{\"configSecret\":{\"encryptedSecret\":\"cvNVJ4D7fNGm59Uh3Tlhi0W3pToDxl7vykUdZaxEuEPGUMdmu7G+xTEfl0CcC2R9aWl6FgrzmX5x+qCGNElUi/PBA40EGjEU\"}}},{\"alias\":\"master2\",\"passphrase\":{\"configSecret\":{\"encryptedSecret\":\"gzXRjE4AoZSW8GQhcqbw0shQfwWGy5O8alNDqXaqTUzyUnLBVUXLwhv+uAEZzecQixiaEaNtfOvuftYW9NjwwRyf6m4KZIag\"}},\"salt\":{\"configSecret\":{\"encryptedSecret\":\"flvAnf5G/aRmDIaahXHjftEpc0odufhzZsuiipxtQTFV+pccB+RHnIqle26jjDlDUBfn3HPbVezKhpRHL/NP5M+cX/rU15DE\"}}}]},\"retrying\":{\"maxAttempts\":{\"default\":3},\"waitBetweenMills\":{\"default\":[200]}}}",
  "schemaVersion": {
    "major": 1,
    "minor": 0
  },
  "version": 0
}'
```